src/Context/ProductsProvider.jsx

```diff
@@ -0,0 +1,30 @@
+import axios from "axios";
+import { createContext, useContext, useEffect, useState } from "react";
+
+const productContext = createContext([]);
+
+const ProductProvider = ({ children }) => {
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    const getproducts = async () => {
+      try {
+        const response = await axios.get("/api/products");
+        setProducts(response.data.products);
+      } catch (error) {
+        console.log(error);
```

Comment:
just a suggestion , we give some message with error , will be helpful while debugging